torch/testing/_internal/opinfo/definitions/signal.py

```diff
@@ -0,0 +1,275 @@
+import random
+import unittest
+from functools import partial
+
+from itertools import product
+from typing import List
+
+import torch
+from torch.testing._internal.common_dtype import all_types_and
+from torch.testing._internal.common_utils import TEST_SCIPY
+from torch.testing._internal.opinfo.core import (
+    DecorateInfo,
+    ErrorInput,
+    OpInfo,
+    SampleInput,
+)
+from torch.testing._legacy import floating_types_and, floating_types
+
+if TEST_SCIPY:
+    import scipy.signal
+
+
+def sample_inputs_window(op_info, device, dtype, requires_grad, *args, **kwargs):
+    r"""Base function used to create sample inputs for windows.
+
+    For additional required args you should use *args, as well as **kwargs for
+    additional keyword arguments.
+    """
+
+    # Test a window size of length zero and one.
+    # If it's either symmetric or not doesn't matter in these sample inputs.
+    for size in [0, 1]:
+        yield SampleInput(
+            size,
+            *args,
+            device=device,
+            dtype=dtype,
+            requires_grad=requires_grad,
+            **kwargs
+        )
+
+    # For sizes larger than 1 we need to test both symmetric and non-symmetric windows.
+    # Note: sample input tensors must be kept rather small.
+    sizes = [2, 5, 10, 50]
```

Comment:
Are these tensors small enough? And regarding the number of sample inputs, do we want to lower the number of them even further? Say, `[2, 5]`?

Reply:
Done!